Add Muon plugin as submodule for submission

wtlnetwork opened the pull request:
<!--
    READ ALL COMMENTS IN THIS TEMPLATE BEFORE SUBMITTING YOUR PR!

    This template is for adding a new plugin to the store. If you are taking any other action, please start over by creating a new pull request and selecting the appropriate template.

    If you have not already, please read the review and testing page on the wiki.
    https://wiki.deckbrew.xyz/plugin-dev/review-and-testing

    Before submitting, make sure you have done the following:
    - Tested two other plugins and left feedback on their PRs as described below.
    - Replaced REPLACE_WITH_PLUGIN_NAME and REPLACE_WITH_SUMMARY with the appropriate information.
    - Filled out the Task Checklist, including the Yes/No questions.
    - Deleted the unnecessary testing checkbox.
-->

# Muon

<!--
    Include a detailed summary of what the plugin does, attaching images or videos if necessary. If your plugin has similar functionality to another plugin on the store, explain how it differs and why your plugin should be allowed on the store.
-->

Muon is a decky-loader plugin which allows the user to start a WiFi hotspot. The primary purpose for doing so is to enable LAN multiplayer between multiple Steam Decks in environments where Internet connectivity is limited (for example, when staying at a hotel or travelling in a car/bus/train/etc).

We have been working on curating a list of compatible games which work with Muon, details of which can be found in this Steam group:
[Muon Verified](https://steamcommunity.com/groups/muonverified)

The plugin has been tested on both the current stable and beta SteamOS versions (3.7.13 and 3.7.13 Beta 'Flyby'). At the time of writing (5 July 2025), there does not appear to be a SteamOS preview build available. Should one become available, we are happy to test on this.

## Task Checklist

<!--
    For checkboxes, change [ ] to [x] to check the box.
    For Yes/No questions, replace "Yes/No" with "Yes" or "No".
-->

### Developer

- [x] I am the original author or an authorized maintainer of this plugin.
- [x] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin

- [x] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [x] I have verified my plugin is unique or provides more/alternative functionality to a plugin already on the store.

### Backend

- **Yes**: I am using a custom backend other than Python. (The plugin utilises Bash scripts in order to activate/deactivate the hotspot and perform networking tasks)
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library). (The plugin depends on hostapd and dnsmasq in order to perform network operations - namely, activation/deactivation of the hotspot and service DHCP requests for clients. These dependencies are downloaded via Pacman and stored in the plugin data directory. It is not necessary for the SteamOS read-only mode to be disabled in order to run the fetched dependencies, and they are removed on uninstall. We have also validated on the latest SteamOS preview that normal system functionality is not impacted by this implementation)
- **No**: I am using a custom binary that has all of it's dependencies statically linked. (Depending on the definition of custom binary - we have not compiled our own binary for Muon, but we are pulling in binaries via Pacman)

### Community

<!--
    Please submit both of your testing reports before creating your PR. You will need to link the comments you left in a comment on this PR.

    If no plugin additions or updates are ready for testing at this time, then you may ignore this checkbox. You may be asked to test new plugins as they are submitted.

    Your PR will be prioritized lower if you do not test other plugins, but this step is optional.
-->

- [ ] I have tested and left feedback on two other [pull requests][pulls] for new or updating plugins.
- [ ] I have commented links to my testing report in this PR.

## Testing

<!--
    DO NOT FORGET THIS STEP! Otherwise, testers may incorrectly test your plugin.

    If your plugin uses the provided Python backend and React frontend, your plugin must be tested on the Stable or Beta update channel of SteamOS. REMOVE the line with the Preview checkbox below.

    If your plugin uses a custom backend or pre-build binaries without statically linked dependency (ex. glibc), your plugin must be tested on the SteamOS Preview update channel. REMOVE the line with the Stable or Beta checkbox below.

    DO NOT CHECK THIS BOX YOURSELF UNLESS SOMEONE HAS COMMENTED A TESTING REPORT FOR YOUR PLUGIN! This will lead to delays in the review process.
-->

- [ ] Tested by a third party on SteamOS Preview update channel.

[pulls]: https://github.com/steamdeckHomebrew/decky-plugin-database/pulls?q=is%3Apr+is%3Aopen+sort%3Acreated-desc+-status%3Afailure+-draft%3Atrue+-author%3A%40me

wtlnetwork commented:
> Please read the comment in the "Testing" section of the PR template and make sure you've removed the correct one of the following lines.
> 
> * Tested by a third party on SteamOS Stable or Beta update channel.
> * Tested by a third party on SteamOS Preview update channel.

Apologies - this has been amended accordingly.

EMERALD0874 reviewed (changes requested):
Overall, your code quality seems good. Fix the following issues and we can start the testing process!

- Using wget in scripts is considered dangerous by our team and is prohibited. Please use the remote binary functionality to ensure all files downloaded match an expected signature. See an example in Decky Framegen [here](https://github.com/xXJSONDeruloXx/Decky-Framegen/blob/033e7f28d62103247e93652d512f64300c656a00/package.json).
- If you aren't using /py_modules/, please remove the folder.
- Please use GitHub's releases feature for new plugin releases, not a dedicated folder. /release/ should be removed.

wtlnetwork commented:
> Overall, your code quality seems good. Fix the following issues and we can start the testing process!
> 
> * Using wget in scripts is considered dangerous by our team and is prohibited. Please use the remote binary functionality to ensure all files downloaded match an expected signature. See an example in Decky Framegen [here](https://github.com/xXJSONDeruloXx/Decky-Framegen/blob/033e7f28d62103247e93652d512f64300c656a00/package.json).
> * If you aren't using /py_modules/, please remove the folder.
> * Please use GitHub's releases feature for new plugin releases, not a dedicated folder. /release/ should be removed.

Thanks for your feedback. I absolutely understand your concerns regarding the use of wget, and I have amended the code to use remote binaries as suggested. I have also removed the py_modules folder and the releases folder. Hopefully this addresses your concerns - please do let me know if anything else comes up.

TrainDoctor commented:
> > Overall, your code quality seems good. Fix the following issues and we can start the testing process!
> > 
> > * Using wget in scripts is considered dangerous by our team and is prohibited. Please use the remote binary functionality to ensure all files downloaded match an expected signature. See an example in Decky Framegen [here](https://github.com/xXJSONDeruloXx/Decky-Framegen/blob/033e7f28d62103247e93652d512f64300c656a00/package.json).
> > * If you aren't using /py_modules/, please remove the folder.
> > * Please use GitHub's releases feature for new plugin releases, not a dedicated folder. /release/ should be removed.
> 
> Thanks for your feedback. I absolutely understand your concerns regarding the use of wget, and I have amended the code to use remote binaries as suggested. I have also removed the py_modules folder and the releases folder. Hopefully this addresses your concerns - please do let me know if anything else comes up.

Please make sure to push these changes as this repo will not ingest them by default.

wtlnetwork commented:
> > > Overall, your code quality seems good. Fix the following issues and we can start the testing process!
> > > 
> > > * Using wget in scripts is considered dangerous by our team and is prohibited. Please use the remote binary functionality to ensure all files downloaded match an expected signature. See an example in Decky Framegen [here](https://github.com/xXJSONDeruloXx/Decky-Framegen/blob/033e7f28d62103247e93652d512f64300c656a00/package.json).
> > > * If you aren't using /py_modules/, please remove the folder.
> > > * Please use GitHub's releases feature for new plugin releases, not a dedicated folder. /release/ should be removed.
> > 
> > 
> > Thanks for your feedback. I absolutely understand your concerns regarding the use of wget, and I have amended the code to use remote binaries as suggested. I have also removed the py_modules folder and the releases folder. Hopefully this addresses your concerns - please do let me know if anything else comes up.
> 
> Please make sure to push these changes as this repo will not ingest them by default.

Thanks for the heads-up - I'm not the most proficient in Git, so I missed that one. Hopefully it's all updated correctly now.

EMERALD0874 reviewed (approved):
LGTM for testing

EMERALD0874 reviewed (changes requested):
Please use the /defaults/ folder to store scripts. This will place them in the root of your plugin directory. You can add subfolders if you want them somewhere else. [Here's](https://github.com/xXJSONDeruloXx/Decky-Framegen/tree/8b3d4def33766f7306d8606ccf4b5e4425a5738a/defaults) an example from Decky Framegen on where they store scripts and [here's](https://github.com/xXJSONDeruloXx/Decky-Framegen/blob/8b3d4def33766f7306d8606ccf4b5e4425a5738a/main.py#L173) an example of how to get the path.

wtlnetwork commented:
Thanks for the heads-up - I'll get this updated ASAP.

wtlnetwork commented:
I have moved the scripts to /defaults/assets now, and updated main.py accordingly. Am I right in thinking that the contents of /defaults/ are copied into the root plugin directory as part of the process for compiling the package for the store?

Up until now, I have been compiling builds manually by adding files to a zip. Looking at the release packages for Framegen, they seem to put the contents of /defaults/assets into /assets in their zips. The main.py reflects this placement (i.e. that the contents of /defaults/assets will end up in /assets) in line with the provided example.

EMERALD0874 commented:
> Am I right in thinking that the contents of /defaults/ are copied into the root plugin directory as part of the process for compiling the package for the store?
Correct.

EMERALD0874 reviewed (approved):
LGTM for testing. @wtlnetwork, please download the plugin from the store, as well, and make sure that the plugin works as intended. :)

wtlnetwork commented:
> Missing Dependencies gets stuck at Installing…
> 
> After checking the logs and install_dependencies.sh, it seems that the folder structure changes (backend/src) between this test version and your previous 0.1.6 dev version broke the install; specifically at line 32 (../../bin/*.pkg.tar.zst)

Thanks for flagging - I think I know what's happened here. I think I put the package path relative to /defaults/assets instead of /assets (where /defaults/assets is copied to when deployed to the store). I'll get a fix rolled out ASAP.

wtlnetwork commented:
I think I see what's happened. Before, we were setting up the sysext in /defaults/extensions, but as this folder does not exist in the storefront version, the script was hanging when trying to set the current working directory to that folder for the subprocess execution. I have amended the code to set up the sysext in /assets instead, and updated the submodule ref accordingly.

If this is not the preferred way to handle sysext modules, another option would be to amend the script to make sure that /defaults/extensions exists before executing the command (or using a different folder) - I'm not sure if there is a standard location for sysext modules that should be used.

theCapypara commented:
# Plugin Testing Report
<!--
    Make sure you replace all text with "REPLACE_WITH" in it. You can
    check for text that needs updating by searching for "REPLACE_WITH"
    in a text editor or web browser before submitting.
-->

## Installed Plugins
<!--
    Post all currently installed plugins.
    Include their versions below this comment block. For example...
    - Free Loader - 1.3.0-6f8f352
    - Decky Terminal - 0.4.1
-->
- SteamGridDB - 1.6.0
- Muon - 0.1.6.2-7dd1245
- Syncthing - 0.2.4
## Specifications
<!--
    Post your SteamOS, Steam, and Decky versions below this comment block.
    For example...
    - SteamOS 3.6.20_20241030.1 (Stable OR Beta OR Preview OR Main)
    - Steam 1733265492 (Stable OR Beta)
    - Decky 3.0.5 (Stable OR Pre-Release)
-->
- NixOS 25.11.20250708.9807714 (Xantusia) with Jovian 0d09755fe1df1886e5f07dbb16f7ed373080f86a
- Steam 1751405894 Stable
- Decky 3.1.8 Stable

## Issues

It says "Error Failed to initialize settings" and when I hit the "Start" button, it still says "Hotspot stopped"., even though I get a toast saying "Hotspot started". As far as I can tell no Hotspot is started. Note that I'm not using SteamOS (see above).

## Summary
<!--
    Leave a brief summary of how you tested the plugin.
    Please include your experience using it below this comment block.
    For example, "Tested by installing themes and using the...
    new theme features in an example provided by the developers.
    Worked as expected".
-->

The plugin did not work for me.

wtlnetwork commented:
> Log:
> 
> You should probably not asume `/bin/bash` exists. Use `/usr/bin/env bash` instead.

Thanks for flagging this - I will make the necessary changes as soon as possible (hopefully by the end of today).

dafta commented:
# Plugin Testing Report
<!--
    Make sure you replace all text with "REPLACE_WITH" in it. You can
    check for text that needs updating by searching for "REPLACE_WITH"
    in a text editor or web browser before submitting.
-->

## Installed Plugins
<!--
    Post all currently installed plugins.
    Include their versions below this comment block. For example...
    - Free Loader - 1.3.0-6f8f352
    - Decky Terminal - 0.4.1
-->
- ProtonDB Badges - 1.1.0
- Free Loader - 1.4.1
- Moon Deck - 1.10.1
- CSS Loader - 2.1.2
- SteamGridDB - 1.6.0
- KDE Connect - 0.1.0-1
- Pause Games - 1.0.0
- Deck Settings - 1.1.1
- IsThereAnyDeal for Deck - 1.0.4
- DeckMTP - 1.0.4
- MagicBlack - 1.0.4
- Muon - 0.1.6.2-8b25125
- ScreenSaver - 1.1.0
- Decky-Lookup -0.1.0

## Specifications
<!--
    Post your SteamOS, Steam, and Decky versions below this comment block.
    For example...
    - SteamOS 3.6.20_20241030.1 (Stable OR Beta OR Preview OR Main)
    - Steam 1733265492 (Stable OR Beta)
    - Decky 3.0.5 (Stable OR Pre-Release)
-->
- SteamOS 3.7.13_20250630.1 (Stable)
- Steam 1751405894 (Stable)
- Decky 3.1.10 (Stable)

## Issues
**Has the following major blocking issue(s):** Does not seem to work after installing the requirements
**Has the following minor non-blocking issue(s):** None

## Summary
<!--
    Leave a brief summary of how you tested the plugin.
    Please include your experience using it below this comment block.
    For example, "Tested by installing themes and using the...
    new theme features in an example provided by the developers.
    Worked as expected".
-->
I installed the plugin, and first it gave a warning that some dependencies are not installed, namely `hostapd`. There was a button to install the dependencies which seemed to work perfectly, and the plugin now had a screen where you can set the SSID and passphrase for the hotspot and start and stop it. Editing settings seems to work, but when I go and press start hotspot, it shows a notification that says "Hotspot started, SSID: [redacted]", but next to the button to start it still says "Hotspot stopped" in red letters, the button still says start, and I can find no WiFi network with the given name.

TrainDoctor commented:
@wtlnetwork When you have time I'm sure @dafta can provide logs and other information to help you determine the issue. For now, this is considered a blocking report and I will be marking the PR until the mentioned issue is resolved.

wtlnetwork commented:
Thanks. Apologies for the lack of response on my part. My whole computer decided to just fully die on me, so it's taken a few days to get back up and running.

We've had a few reports of the wlan0 device disappearing. At first, we thought it might be down to VPN services (plugins like Zerotier and Tunneldeck) but apparently it's also happening on Decks without these, too. I haven't been able to replicate so far but thanks to your logs, and those on a bug report we've had last night, I hopefully should be able to get to the bottom of this.

dafta commented:
I've got a self-hosted VPN setup using Wireguard, if that helps. I imported it using NetworkManager and I don't use any plugin for it. It's got it's own network device, `wg0`.
